integration-tests/smoke/ccip/ccip_add_chain_test.go

```diff
@@ -0,0 +1,851 @@
+		{
+			Changeset: commonchangeset.WrapChangeSet(ccipcs.UpdateOnRampsDests),
+			Config: ccipcs.UpdateOnRampDestsConfig{
+				UpdatesByChain: onRampDestUpdates(t, newChains, sources, testRouterEnabled),
+				MCMS:           mcmsConfig,
+			},
+		},
+		{
+			Changeset: commonchangeset.WrapChangeSet(ccipcs.UpdateFeeQuoterPricesCS),
+			Config: ccipcs.UpdateFeeQuoterPricesConfig{
+				PricesByChain: feeQuoterPricesByChain(t, newChains, sources),
+				MCMS:          mcmsConfig,
+			},
+		},
+		{
+			Changeset: commonchangeset.WrapChangeSet(ccipcs.UpdateFeeQuoterDests),
+			Config: ccipcs.UpdateFeeQuoterDestsConfig{
+				UpdatesByChain: feeQuoterDestUpdates(t, newChains, sources),
+				MCMS:           mcmsConfig,
+			},
+		},
+		{
+			Changeset: commonchangeset.WrapChangeSet(ccipcs.UpdateRouterRamps),
+			Config: ccipcs.UpdateRouterRampsConfig{
+				TestRouter:     testRouterEnabled,
+				UpdatesByChain: routerOnRampUpdates(t, newChains, sources),
+				MCMS:           mcmsConfig,
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	return e
+}
+
+// setupChain will deploy the ccip chain contracts to the provided chains.
+// Based on the flags provided, it will also deploy the jobs and home chain contracts.
+// mcmsEnabled should be set to true if the home chain contracts have been transferred to MCMS.
+func setupChain(t *testing.T, e ccipcs.DeployedEnv, tEnv ccipcs.TestEnvironment, chains []uint64, deployJobs, deployHomeChain, mcmsEnabled bool) ccipcs.DeployedEnv {
+	e = ccipcs.AddCCIPContractsToEnvironment(
+		t,
+		chains,
+		tEnv,
+		deployJobs,
+		deployHomeChain,
+		mcmsEnabled,
+	)
+
+	// Need to update what the RMNProxy is pointing to, otherwise plugin will not work.
+	var err error
+	e.Env, err = commonchangeset.ApplyChangesets(t, e.Env, e.TimelockContracts(t), []commonchangeset.ChangesetApplication{
+		{
+			Changeset: commonchangeset.WrapChangeSet(ccipcs.SetRMNRemoteOnRMNProxy),
+			Config: ccipcs.SetRMNRemoteOnRMNProxyConfig{
+				ChainSelectors: chains,
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	return e
+}
+
+// assertChainWiringInbound checks that the new chain has the existing chains enabled as sources.
+// It only checks the inbound wiring on the new chain.
+// It doesn't check that the existing chains have the new chain enabled as a dest.
+func assertChainWiringInbound(
+	t *testing.T,
+	state ccipcs.CCIPOnChainState,
+	newChain uint64,
+	existingChains []uint64,
+	testRouterEnabled bool,
+) {
+	for _, existingChain := range existingChains {
+		var rtr *router.Router
+		if testRouterEnabled {
+			rtr = state.Chains[newChain].TestRouter
+		} else {
+			rtr = state.Chains[newChain].Router
+		}
+
+		// check that the offRamp has the existing chain enabled as a source.
+		dcc, err := state.Chains[newChain].OffRamp.GetSourceChainConfig(&bind.CallOpts{
+			Context: tests.Context(t),
+		}, existingChain)
+		require.NoError(t, err)
+		require.Equal(t, rtr.Address(), dcc.Router)
```

Comment:
let's also check the onRamp is set correctly to source OnRamp and whether it's enabled

Reply:
That's a check on the `sourceChains` right? That's already done in `assertChainWiringOutbound` I believe.

Reply:
is `assertChainWiringOutbound` asserting the onRamp address on GetSourceChainConfig?

Reply:
Ah no it doesn't do that cross-chain check, but I can add it.